tom/packages.py

```diff
@@ -0,0 +1,96 @@
+import re
+import json
+import requests
+import collections
+import datetime
+import hashlib
+import urllib.request
+import logging as log
+from tom.git import GitRepo
+from tom.utils import pretty
+
+class PackageMapper():
+    """Class responsible for updating packages_mapping.json file
+    Currently it's tailored for cfengine needs.
+    """
+
+    def __init__(self, github, slack, dispatcher, username):
+        self.github = github
+        self.slack = slack
+        self.username = username
+        dispatcher.register_command(
+                keyword='packages_mapping',
+                callback=lambda branches: self.run(branches),
+                parameter_name='branches',
+                short_help='Add packages to packages_mapping.json',
+                long_help='Scan releases.json for last release of given branches and add all of their packages to packages_mapping file')
+
+    def run(self, branches):
+        """Update the packages_mapping file and create PR"""
+        # prepare repo
+        repo_name = 'system-testing'
+        upstream_name = 'cfengine'
+        local_path = "../" + repo_name
+        repo = GitRepo(local_path, repo_name, upstream_name, self.username, "master")
+        timestamp = re.sub('[^0-9-]', '_', str(datetime.datetime.today()))
+        new_branchname = 'packages_mapping-{}'.format(timestamp)
+        repo.checkout(new_branchname, new=True)
+        # load current file
+        packages_mapping_file_path = 'deployment_tests/packages_mapping.json'
+        packages_mapping = json.loads(repo.get_file(packages_mapping_file_path), object_pairs_hook=collections.OrderedDict)
```

Comment:
Please split this into multiple lines with exceptions being handled gracefully.

Reply:
exceptions if there is no `packages_mapping.json` file and if it contains invalid json? okay...

Reply:
Not a big deal in this case, I originally confused it with the step that fetches and parses the `releases.json` file and stuff. But still nice to at least have the `repo.get_file()` and `json.loads()` done in two steps on two lines, I think.